### PR TITLE
Add TransposeMatMul transformation to MOC

### PR DESCRIPTION
**Pull request by mateusztabaka**
Transformation fuses Transpose on first or second MatMul's input and sets MatMul's transpose_a/transpose_b accordingly.

TransposeMatMul is already part of SmartReshape, but it can be added to MOCTransformations as well so native models that are don't use reshape can benefit from that.

Ticket: CVS-118908

some GNA tests were disabled, ticket: CVS-123306

### CODE REVIEW

**Review by praasz — APPROVED** [review]

File: src/common/transformations/src/transformations/common_optimizations/moc_transformations.cpp [hunk 1]
```diff
@@ -165,11 +166,13 @@ bool ov::pass::MOCTransformations::run_on_model(const std::shared_ptr<ov::Model>
 
     auto transpose_sinking = manager.register_pass<ov::pass::GraphRewrite>();
     ADD_MATCHER(transpose_sinking, TransposeSinking)
-
     // SplitSqueezeConcatFusion should work in same GraphRewrite as TransposesSinking,
     // because it replaces pattern that may contain Transposes which must be optimized before
     // the transformation and it also inserts Transpose that can be optimized by TransposeSinking
     ADD_MATCHER(transpose_sinking, SplitSqueezeConcatFusion)
+
+    REGISTER_PASS(manager, TransposeMatMul)
```
Comment (itikhono) [inline src/common/transformations/src/transformations/common_optimizations/moc_transformations.cpp:174]: minor: maybe it's better to move it to `common_fusions` transformations?

**Review by itikhono — APPROVED** [review]

File: src/common/transformations/src/transformations/common_optimizations/moc_transformations.cpp (hunk 1 above)
Comment (itikhono) [reply src/common/transformations/src/transformations/common_optimizations/moc_transformations.cpp:174]: let's do it in the next PR

**Comment by itikhono** [comment]
@sirzabek could you take a look at the changes in this file src/plugins/intel_gna/src/transformations/handle_transposes_around_matmul.cpp?

File: src/plugins/intel_gna/src/transformations/handle_transposes_around_matmul.cpp [hunk 2]
```diff
@@ -150,11 +150,13 @@ HandleTransposeBeforeMatMul::HandleTransposeBeforeMatMul() {
             return false;
         }
 
-        auto matmul_node = matmul_iter->second.get_node_shared_ptr();
+        auto matmul_node = std::dynamic_pointer_cast<ngraph::opset8::MatMul>(matmul_iter->second.get_node_shared_ptr());
+        if (!matmul_node)
+            return false;
```
Comment (sirzabek) [inline src/plugins/intel_gna/src/transformations/handle_transposes_around_matmul.cpp:155]: Why is this required here? We shouldn't be here with a layer other than MatMul, because of pattern_map search above, right?

File: src/plugins/intel_gna/src/transformations/handle_transposes_around_matmul.cpp [hunk 3]
```diff
@@ -150,11 +150,13 @@ HandleTransposeBeforeMatMul::HandleTransposeBeforeMatMul() {
...
         auto transpose_reshape_it = pattern_map.find(transpose1);
         if (transpose_reshape_it != std::end(pattern_map)) {
             ReplaceTransposeWithReshape(transpose_reshape_it->second.get_node_shared_ptr());
-        } else {
+        } else if (!matmul_node->get_transpose_a()) {
```
Comment (sirzabek) [inline src/plugins/intel_gna/src/transformations/handle_transposes_around_matmul.cpp:159]: Could you please explain why is this logic change required?

Comment (mateusztabaka) [reply src/plugins/intel_gna/src/transformations/handle_transposes_around_matmul.cpp]: I disabled the tests for now.

Comment (itikhono) [reply src/plugins/intel_gna/src/transformations/handle_transposes_around_matmul.cpp]: ticket number: CVS-123306

**Review by sirzabek — APPROVED** [review]
